### ISSUE TITLE
[Question] Is there a way to use custom gene sets? 

### ISSUE THREAD

**Issue by jolespin** [zqfang/GSEApy#60]
Is there a way to use custom gene sets?  For example, what if I had the following data:
```python
# Gene list
gene_list = [
"gene_A", 
"gene_B", 
...
"gene_xyz"
]
```
and then a bunch of gene sets like this:
```python
gene_sets = {
"gene_set_1": ["gene_A", "gene_B", ...],
"gene_set_2":["gene_B", gene_C", ...], 
...
"gene_set_100":["gene_A", "gene_T", ...]
}
```

**Is there still a way to run this tool to figure out which gene sets are enriched?** 

I usually deal with microbiome datasets with de-novo called ORFs from prodigal so there are no gene ids that would be useable.  A lot of my friends in cancer labs always talk about GSEA but I don't have the types of IDs that they use.  Though, I still have "gene sets" that I could use so I feel like it could still apply.

**Comment by zqfang** [comment]
Yes. GSEApy definitely supports custom gene sets, except ``enrichr`` module.  So, inside python console, run your cmd like this:
 
    >>> import gesapy
    >>> gseapy.prerank(rnk=cor_vector, gene_sets=gene_sets,......)

**Comment by jolespin** [comment]
Oh wow that is awesome! 

So it works with dictionaries as well without the need of an external gmt file?

One more question kind of related. I just read the gsea paper and it referenced the correlations between categorical phenotypes and genes.  Is there a specific way to do this or is it literally just a vector gene expression values for each sample and a vector of 0s and 1s relating to phenotype with Pearson/spearman/biweight correlations? 

If this extra question is out of scope or worthy of a close and separate issue please let me know.

**Comment by zqfang** [comment]
Yes, it works without the need of gmt file. For the next question, You could just provide a gene expression values (TPM,FPKM, normalized counts) for each sample. You could also define a custom metric (e.g. foldchange between two phenotype) by calling ``Prerank`` module. GSEA prefers continuous values input, instead of 0s and 1s.

**Comment by RBenRocks** [comment]
Is there any chance you can implement this on the enrichr module? I am currently unable to provide gene expression values, and just want to use a gene list as input as well as my defined gene sets. Thanks so much in advance?

**Comment by zqfang** [comment]
If you mean the custom gene sets input for enrichr module, then NO. But the next release will support custom file input for ``enrichr`` module. This function are still in development, you could give a try by installing the latest PR. @RBenRocks

**Comment by zqfang** [comment]
The request are implemented in v0.9.9. Closed now
